src/renderer/components/Input/InputDelegate.vue

```diff
@@ -0,0 +1,362 @@
+      }
+
+      return null
+    },
+
+    hasSuggestions () {
+      return !_.isEmpty(this.suggestions)
+    },
+
+    invalid () {
+      return this.$v.model.$dirty && (this.isInvalid || !!this.error)
+    },
+
+    model: {
+      get () {
+        return this.dropdownValue || this.inputValue
+      },
+      set (value) {
+        this.updateInputValue(value)
+        this.$emit('input', value)
+      }
+    },
+
+    suggestions () {
+      if (!this.currentProfile) return []
+
+      const delegates = _.map(this.delegates, (object) => {
+        const delegate = {
+          name: null,
+          username: object.username,
+          address: object.address,
+          publicKey: object.publicKey
+        }
+
+        delegate.name = `${truncate(object.username, 25)} (${this.wallet_truncate(object.address)})`
+
+        return delegate
+      })
+
+      const results = _.orderBy(delegates, (object) => {
+        return object.name || object.address.toLowerCase()
+      })
+
+      return results.reduce((delegates, delegate, index) => {
+        Object.values(delegate).forEach(prop => {
+          if (_.includes(prop.toLowerCase(), this.inputValue.toLowerCase())) {
+            delegates[delegate.username] = delegate.name
+          }
+        })
+
+        return delegates
+      }, {})
+    },
+
+    suggestionsKeys () {
+      return new Cycled(Object.keys(this.suggestions))
+    }
+  },
+
+  watch: {
+    value (val) {
+      this.updateInputValue(val)
+    },
+
+    isFocused () {
+      if (this.isFocused && this.hasSuggestions) {
+        this.openDropdown()
+      }
+    },
+
+    inputValue () {
+      this.dropdownValue = null
+      if (this.isFocused && this.hasSuggestions) {
+        this.openDropdown()
+      }
+    }
+  },
+
+  mounted () {
+    if (this.value) {
+      this.updateInputValue(this.value)
+    }
+  },
+
+  methods: {
+    getHelperText () {
+      return (!this.$refs.dropdown || !this.$refs.dropdown.isOpen) ? this.helperText : ''
+    },
+
+    blur () {
+      this.$refs.input.blur()
+    },
+
+    focus () {
+      this.$refs.input.focus()
+    },
+
+    onBlur (evt) {
+      // Verifies that the element that generated the blur was a dropdown item
+      if (evt.relatedTarget) {
+        const classList = evt.relatedTarget.classList || []
+        const isDropdownItem = _.includes(classList, 'MenuDropdownItem__button')
+
+        if (!isDropdownItem) {
+          this.closeDropdown()
+        }
+      } else if (this.$refs.dropdown.isOpen) {
+        this.closeDropdown()
+      }
+
+      this.isFocused = false
+
+      // If the user selects a suggestion and leaves the input
+      if (this.dropdownValue) {
+        this.onEnter()
+      }
+    },
+
+    onDropdownSelect (value) {
+      this.model = value
+      this.$nextTick(() => this.closeDropdown())
+    },
+
+    onFocus () {
+      this.isFocused = true
+      this.$emit('focus')
+    },
+
+    onEnter () {
+      if (!this.dropdownValue) return
```

Comment:
Try to keep if statements consistent with curly brackets as it helps with readability